### PR TITLE
release-003

### PR DESCRIPTION
**Pull request by tahb**
- add database foreign key constraints for better data integrity
- Contentful entry requests use a new Redis read cache
- service name included in the header
- refactor name of "Plan" to "Journey"
- refactor name of "Question" to "Step"
- refactor redis caching into reusable class
- users can be shown a step in the journey that is only static content

### CODE REVIEW

**Comment by erbridge** [comment]
Should this target `main`?

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,8 @@ The format is based on [Keep a Changelog 1.0.0].
 
 ## [Unreleased]
 
+## [release-003] - 2020-12-03
```
Comment (erbridge) [inline CHANGELOG.md:9]: Should probably update this date.

File: CHANGELOG.md [hunk 2]
```diff
@@ -34,7 +36,8 @@ Contentful fixture
 - Contentful can redirect users to preview endpoints
 - users can be asked to answer a long text question
 
-[unreleased]: https://github.com/DfE/DFE-Digital/buy-for-your-school/compare/release-001...HEAD
+[unreleased]: https://github.com/DfE/DFE-Digital/buy-for-your-school/compare/release-003...HEAD
+[release-003]: https://github.com/DfE/DFE-Digital/buy-for-your-school/compare/release-002...release-003
```
Comment (erbridge) [inline CHANGELOG.md:40]:
```suggestion
[unreleased]: https://github.com/DFE-Digital/buy-for-your-school/compare/release-003...HEAD
[release-003]: https://github.com/DFE-Digital/buy-for-your-school/compare/release-002...release-003
```

**Comment by tahb** [comment]
[We merge into develop and then manually merge develop into master when ready](https://github.com/DFE-Digital/buy-for-your-school/blob/develop/doc/release-process.md#2-confirm-the-release-and-review-the-pull-request).

**Comment by tahb** [comment]
@erbridge thanks for the comments, I've updated the date and changed the links.

**Review by erbridge — APPROVED** [review]

**Pull request merged**